binstar_client/inspect_package/conda.py

```diff
@@ -75,18 +75,21 @@ def get_subdir(index):
 
 def inspect_conda_package(filename, fileobj, *args, **kwargs):
 
-    tar = tarfile.open(filename, fileobj=fileobj, mode="r|bz2")
-    recipe = {}
-    for info in tar:
-        if info.name == 'info/index.json':
-            index = tar.extractfile(info)
-            index = json.loads(index.read().decode())
-        elif info.name == 'info/recipe.json':
-            try:
-                recipe = tar.extractfile(info)
-                recipe = json.loads(recipe.read().decode())
-            except KeyError:
-                recipe = {}
+    index, recipe = None, None
+
+    with tarfile.open(filename, fileobj=fileobj, mode="r|bz2") as tar:
+        for info in tar:
+            if info.name == 'info/index.json':
+                index = tar.extractfile(info)
+                index = json.loads(index.read().decode())
+            elif info.name == 'info/recipe.json':
+                try:
+                    recipe = tar.extractfile(info)
+                    recipe = json.loads(recipe.read().decode())
+                except KeyError:
+                    recipe = None
```

Comment:
This `try/except` around `extractfile` is also unnecessary - that was needed if `info/recipe.json` did not exist in the package - but since we're inside this branch, the file definitely exists.